applications/GeoMechanicsApplication/tests/cpp_tests/test_time_loop_executor.cpp

```diff
@@ -0,0 +1,356 @@
+{
+    auto result = TimeStepEndState{};
+    result.convergence_state = TimeStepEndState::ConvergenceState::converged;
+    return result;
+}
+
+std::shared_ptr<DummySolverStrategy> RunFixedCycleTimeLoop(std::size_t WantedNumOfCyclesPerStep)
+{
+    TimeLoopExecutor executor;
+    executor.SetTimeIncrementor(std::make_unique<FixedCyclesTimeIncrementor>(WantedNumOfCyclesPerStep));
+    auto solver_strategy = std::make_shared<DummySolverStrategy>(TimeStepEndState::ConvergenceState::converged);
+    executor.SetSolverStrategyTimeStepExecutor(solver_strategy);
+    const auto step_states = executor.Run(TimeStepEndState{});
+
+    return solver_strategy;
+}
+
+}
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(TimeLoopReturnsPerformedStatesAfterRunningAnAlwaysConvergingSolverStrategy, KratosGeoMechanicsFastSuite)
+{
+    TimeLoopExecutor executor;
+    const auto increments = std::vector<double>{1.1, 1.2, 1.3, 1.4};
+    executor.SetTimeIncrementor(std::make_unique<PrescribedTimeIncrementor>(increments));
+    executor.SetSolverStrategyTimeStepExecutor(
+            std::make_unique<DummySolverStrategy>(TimeStepEndState::ConvergenceState::converged));
+    TimeStepEndState start_state;
+    start_state.convergence_state = TimeStepEndState::ConvergenceState::converged;
+    const auto step_states = executor.Run(start_state);
+
+    KRATOS_EXPECT_EQ(increments.size(), step_states.size());
+    KRATOS_EXPECT_TRUE(std::all_of(step_states.begin(), step_states.end(), [](const auto& step_state) { return step_state.Converged(); }))
+}
+
+KRATOS_TEST_CASE_IN_SUITE(TimeLoopReturnsOneNonConvergedPerformedStatesAfterRunningAnNeverConvergingSolverStrategy, KratosGeoMechanicsFastSuite)
+{
+    TimeLoopExecutor executor;
+    const auto increments = std::vector<double>{1.1, 1.2, 1.3, 1.4};
+    executor.SetTimeIncrementor(std::make_unique<PrescribedTimeIncrementor>(increments));
+    executor.SetSolverStrategyTimeStepExecutor(
+            std::make_unique<DummySolverStrategy>(TimeStepEndState::ConvergenceState::non_converged));
+    TimeStepEndState start_state;
+    start_state.convergence_state = TimeStepEndState::ConvergenceState::converged;
+    const auto step_states = executor.Run(start_state);
+
+    KRATOS_EXPECT_EQ(1, step_states.size());
+    KRATOS_EXPECT_TRUE(step_states[0].NonConverged())
+}
+
+KRATOS_TEST_CASE_IN_SUITE(TimeLoopReturnsEndTimesAfterRunningAnAlwaysConvergingSolverStrategy, KratosGeoMechanicsFastSuite)
+{
+    TimeLoopExecutor executor;
+    const auto increments = std::vector<double>{1.1, 1.2, 1.3, 1.4};
+    executor.SetTimeIncrementor(std::make_unique<PrescribedTimeIncrementor>(increments));
+    executor.SetSolverStrategyTimeStepExecutor(
+            std::make_unique<DummySolverStrategy>(TimeStepEndState::ConvergenceState::converged));
+    TimeStepEndState start_state;
+    start_state.convergence_state = TimeStepEndState::ConvergenceState::converged;
+    const auto step_states = executor.Run(start_state);
```

Comment:
```suggestion
    TimeStepEndState start_state;
    start_state.convergence_state = TimeStepEndState::ConvergenceState::converged;
    const auto step_states = executor.Run(MakeConvergedStepState());
```

Reply:
Done ( with removing the start_state too.